rfcs/0051-limit-zkapp-cmds-per-block.md

```diff
@@ -0,0 +1,161 @@
+## Summary
+
+During the ITN stress testing it was noticed that daemon's memory
+consumption tends to increase dramatically after a block containing a
+large number of zkApp commands. Before appropriate optimizations can
+be developed, we need a temporary solution to prevent nodes crashing
+due to insufficient memory. The idea is to limit the number of zkApp
+commands that can be included in any single block.
+
+## Motivation
+
+By limiting the number of zkApp commands going into blocks we avoid
+the aforementioned issue until a proper solution can be devised and
+implemented. The root cause of the issue is that proofs contained
+within these commands are stored in the scan state and tend to occupy
+a lot of space. Fixing these storage issues won't affect the
+protocol, so ideally we want a workaround that doesn't affect the
+protocol either, so that at the convenient time we can turn it off
+without making a fork.
+
+## Detailed design
+
+Since the solution should not affect the protocol, it should be
+implemented at the mempool/block producer boundary. In the mempool
+there is `transactions` function, which returns a sequence of
+transactions from the mempool in the order of decreasing transaction
+fees. The block producer module just takes a number of transactions
+from that sequence and includes them. It is relatively easy to limit
+the number of zkApp commands returned by this functions.  ZkApp
+commands above a certain limit would simply be discarded (from the
+sequence, but not from the mempool).
+
+The exact number of zkApps allowed in each block should be set
```

Comment:
It should not be dynamically configurable at mainnet though, so we need to take care with how we are exposing dynamic configuration, and how we are handling it internally based on network id.